mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -673,9 +675,11 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         itemFactors = computeFactors(userFactors, userOutBlocks, itemInBlocks, rank, regParam,
           userLocalIndexEncoder, solver = solver)
         if (shouldCheckpoint(iter)) {
+          val deps = itemFactors.dependencies
           itemFactors.checkpoint()
           itemFactors.count() // checkpoint item factors and cut lineage
           deletePreviousCheckpointFile()
+          ALS.cleanDependencies(sc, deps)
```

Comment:
minor but let's move this up one line to be consistent with the implicit case above.